init-tools.sh

```diff
@@ -22,18 +22,14 @@ fi
 
 if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
 
-if [ -d "$DotNetBuildToolsDir" ]; then
+if [ "$DotNetBuildToolsDir" != "" ]; then
     echo "Using tools from '$DotNetBuildToolsDir'."
     ln -s "$DotNetBuildToolsDir" "$__TOOLRUNTIME_DIR"
 
     if [ ! -e "$__DOTNET_CMD" ]; then
         echo "ERROR: Ensure that $DotNetBuildToolsDir contains the .NET Core SDK at $__DOTNET_PATH"
         exit 1
     fi
-
-    echo "Done initializing tools."
-    mkdir -p "$(dirname "$__BUILD_TOOLS_SEMAPHORE")" && touch $__BUILD_TOOLS_SEMAPHORE
-    return #return instead of exit because this script is inlined in other scripts which we don't want to exit
```

Comment:
We have not checked that build tools are installed, we shouldn't early out.

@weshaggard This assumes this section is intended for bootstrap use.  If not we can make this code conditional.